library/multipicker/src/main/java/im/vector/lib/multipicker/Picker.kt

```diff
@@ -115,6 +116,14 @@ abstract class Picker<T> {
                 }
             }
         }
-        return selectedUriList.onEach { context.grantUriPermission(context.applicationContext.packageName, it, Intent.FLAG_GRANT_READ_URI_PERMISSION) }
+        selectedUriList.forEach { uri ->
+            try {
+                context.grantUriPermission(context.applicationContext.packageName, uri, Intent.FLAG_GRANT_READ_URI_PERMISSION)
+            } catch (e: SecurityException) {
+                // Handle the exception, e.g., log it or notify the user
+                Timber.w("Picker", "Failed to grant URI permission for $uri: ${e.message}")
```

Comment:
If the permission is not granted, I guess the issue will appear later, when the app will try to access the file? What will happen in this case?

Reply:
I use MDM, and having two containers, Personal space and Work space. Space is restricted with certain policies for accessing file in between. 

Even though the  explicit call `grantUriPermission` fails, the intent with flag `Intent.FLAG_GRANT_READ_URI_PERMISSION ` succeed.

This ability to share files using intents with FLAG_GRANT_READ_URI_PERMISSION while grantUriPermission fails is likely a result of MDM policies that enforce strict separation between work and personal profiles. These policies are designed to protect user data while still allowing some level of interaction through standard Android intent mechanisms